src/core/localize.rkt

```diff
@@ -164,52 +177,84 @@
     (define exacts (list->vector (apply subexprs-fn pt)))
     (define actuals (apply actual-value-fn pt))
 
-    (for ([expr (in-list exprs-list)]
+    (for ([spec (in-vector spec-vec)]
+          [expr (in-list exprs-list)]
           [root (in-vector roots)]
           [exact (in-vector exacts)]
           [actual (in-vector actuals)]
           [expr-idx (in-naturals)])
-      (define err
+      (define true-err
+        (if true-err? ;; ??? Whats the default values for true error literal, variable approx and if?
+            (match (vector-ref nodes root)
+              [(? literal?) 0]
+              [(? variable?) 0]
+              [(approx aprx-spec impl) exact] ;; TODO not sure what to do here.
+              [`(if ,c ,ift ,iff) 0]
+              [(list f args-roots ...)
+               ;; Find the index of the variables we need to substitute.
+               (match exact
+                 [`+nan.0 `+nan.0]
+                 [`-nan.0 `-nan.0]
+                 [`+inf.0 `+inf.0]
+                 [`-inf.0 `-inf.0]
+                 [value
+                  ; __exact double underscore to avoid conflicts with user provided
+                  ; variables. Could use name mangling long term.
+                  (define modifed-vars (append all-vars `(__exact)))
+                  (define true-error-expr (list `(- ,spec __exact)))
+                  (define diffMachine
+                    (rival-compile true-error-expr modifed-vars (list flonum-discretization)))
+                  (define inputs (map (representation-repr->bf our_repr) (append pt (list exact))))
+                  ;; ??? Is this always length 1, as we are asking about exact?
```

Comment:
Why are you using `rival` directly here instead of the `real-machine` stuff in `rival.rkt`?